docs/_docs/04_reference/04-config-blocks-and-attributes.md

```diff
@@ -1624,7 +1624,37 @@ unit "vpc" {
 }
 ```
 
-Note that each unit must have a unique name and path within the stack. The values provided in the `values` block will be made available to the unit's configuration as input variables.
+Note that each unit must have a unique name and path within the stack.
+
+When `values` are specified, generated units will have access to those values via a special `terragrunt.values.hcl` file generated next to the `terragrunt.hcl` file of the unit.
+
+```tree
+terragrunt.stack.hcl
+.terragrunt-stack
+├── vpc
+│   ├── terragrunt.values.hcl
+│   └── terragrunt.hcl
+```
+
+The `terragrunt.values.hcl` file will contain the values specified in the `values` block as top-level attributes:
+
+```hcl
+# .terragrunt-stack/vpc/terragrunt.values.hcl
+
+vpc_name = "main"
+cidr     = "10.0.0.0/16"
+```
+
+The unit will be able to leverage those values via `values` variables.
```

Comment:
Should it be noted that this will only work if the stacks experiment is enabled?

Reply:
Good point.